agent.go

```diff
@@ -853,12 +886,15 @@ func (c *controller) handleEpTableEvent(ev events.Event) {
 		nid = event.NetworkID
```

Comment:
mmm... can we? I was hoping so initially, but I'm not sure how we can access the fields of each individual type except through the switch (or equivalent explicit type conversions). It's possible that my understanding of go is wrong, but from what I understand event in each of these separate switch branches has a different type. Unless we were doing C-style unsafe data overlay stuff here, I'm not sure how we can otherwise get to the fields even if they all have the same names, types and order declaration. Please do let me know if I'm wrong.

Reply:
Actually looking at it better this can be compacted as:
```
switch event := ev.(type) {
	case networkdb.CreateEvent:
	case networkdb.DeleteEvent:
	case networkdb.UpdateEvent:
		nid = event.NetworkID
		eid = event.Key
		value = event.Value
	default:
		logrus.Errorf("Unexpected update service table event = %#v", event)
		return
	}
```

both the 3 events are actually defined as:
```
// CreateEvent generates a table entry create event to the watchers
type CreateEvent event

// UpdateEvent generates a table entry update event to the watchers
type UpdateEvent event

// DeleteEvent generates a table entry delete event to the watchers
type DeleteEvent event
```
so they are all sharing the same base type:
```
type event struct {
	Table     string
	NetworkID string
	Key       string
	Value     []byte
}
```
make the collapse of the 3 cases possible

Reply:
What you wrote above is legal, but doesn't do what you think it does.  :)  CreateEvent and DeleteEvent cases are empty branches.  So they will not populate nid, eid, and value.  Furthermore, you can't fix that by adding `fallthrough` to those branches because `fallthrough` is specifically forbidden in type switches.   However, since they are all `type event` under the hood, I can compact it with a second type assertion.  Will try this out and repost.

Reply:
Actually, I can't do that because networkdb.event is not an exported type.  :(

Reply:
sorry I wrote it C++ style, and yes because of event is not exported won't work if you don't change the scope. Ok I'm giving up